### PR TITLE
Super keyword.

### PR DESCRIPTION
**Pull request by chriseth**
This allows access to overridden functions even in a diamond-shaped inheritance graph.

### CODE REVIEW

File: libsolidity/GlobalContext.cpp [hunk 1]
```diff
@@ -83,5 +83,14 @@ MagicVariableDeclaration const* GlobalContext::getCurrentThis() const
...
 }
 
+MagicVariableDeclaration const* GlobalContext::getCurrentSuper() const
+{
+	if (!m_superPointer[m_currentContract])
+		m_superPointer[m_currentContract] = make_shared<MagicVariableDeclaration>(
+													"super", make_shared<ContractType>(*m_currentContract, true));
+	return m_superPointer[m_currentContract].get();
+
+}
```
Comment (LianaHus) [inline libsolidity/GlobalContext.cpp]: is this space before "super..." line mistyping?

File: libsolidity/ExpressionCompiler.cpp [hunk 2]
```diff
@@ -360,14 +360,22 @@ void ExpressionCompiler::endVisit(MemberAccess const& _memberAccess)
...
+		else
+		{
+			u256 identifier = type.getFunctionIdentifier(member);
+			if (identifier != Invalid256)
+			{
+				appendTypeConversion(type, IntegerType(0, IntegerType::Modifier::ADDRESS), true);
+				m_context << identifier;
+				break;
+			}
+			// fall-through to "integer" otherwise (address)
```
Comment (LianaHus) [inline libsolidity/ExpressionCompiler.cpp]: I would move this comment before if statement

File: libsolidity/GlobalContext.cpp (hunk 1 above)
Comment (LefterisJP) [inline libsolidity/GlobalContext.cpp]: no it's the unfortunate way that github chooses to show diffs with our code style. Has to do with number of spaces assigned to each tab.

Comment (LianaHus) [inline libsolidity/GlobalContext.cpp]: ah. but in code(->View) was the same

File: libsolidity/GlobalContext.cpp [hunk 3]
```diff
@@ -83,5 +83,14 @@ MagicVariableDeclaration const* GlobalContext::getCurrentThis() const
 
 }
 
+MagicVariableDeclaration const* GlobalContext::getCurrentSuper() const
+{
+	if (!m_superPointer[m_currentContract])
+		m_superPointer[m_currentContract] = make_shared<MagicVariableDeclaration>(
+													"super", make_shared<ContractType>(*m_currentContract, true));
+	return m_superPointer[m_currentContract].get();
+
```
Comment (gavofyork) [inline libsolidity/GlobalContext.cpp]: blank line?

File: libsolidity/GlobalContext.cpp (hunk 1 above)
Comment (chriseth) [inline libsolidity/GlobalContext.cpp]: Sometimes you can add "?ts=4" to the URL in github to get the tab->space conversion we use.

**Comment by chriseth** [comment]
Should be merged after #858 .

**Comment by chriseth** [comment]
Removed a test since it was fragile with respect to the order of functions in the generated code and in the end it did not really test anything anyway.

**Pull request merged**